src/bosh-director/lib/bosh/director/starter.rb

```diff
@@ -1,33 +1,63 @@
 module Bosh::Director
   class Starter
     class << self
-      def start(
-        instance:,
-        agent_client:,
-        update_config:,
-        is_canary: false,
-        wait_for_running: true,
-        logger: Config.logger
-      )
-        logger.info("Running pre-start for #{instance}")
-        agent_client.run_script('pre-start', {})
-
-        logger.info("Starting instance #{instance}")
-        agent_client.start
+      def start(args = {})
```

Comment:
I'm not a huge fan of this pattern and I prefer the previous argument setup. Yes it has a lot of arguments but it's easier to tell what is required vs optional just with the method signature.

Reply:
I totally agree, however the previous argument setup was not accepted by rubocop. The method already had to many arguments before my change, but because I added an argument I had to "fix" the offence. The better solution would probably be a bigger refactoring, but since I'm not that familiar with the codebase yet I have opted for this approach.

Reply:
Bummer, I'll leave it to your discretion then. I've also chosen at times to ignore rubocop if I think it's wrong. You can use `git commit --no-verify` to skip it if you want.